keylime/ima/ima.py

```diff
@@ -472,7 +492,53 @@ def read_allowlist(al_path=None, checksum="", gpg_sig_file=None, gpg_key_file=No
                 f"Checksum of allowlist does not match! Expected {checksum}, Calculated {calculated_checksum}"
             )
 
+    # Return artifacts as a JSON bundle, ready for sending
+    bundle = {
+        "ima_policy": base64.b64encode(alist_bytes).decode(),
+        "key": base64.b64encode(al_key).decode(),
+        "sig": base64.b64encode(al_sig).decode(),
+        "checksum": checksum,
+    }
+
+    return bundle
+
+
+class SignatureValidationError(Exception):
+    def __init__(self, message, code):
+        self.message = message
+        self.code = code
+        super().__init__(self.message)
+
+
+# Reads allowlist bundles sent to the verifier, validates them, and processes them. Returns an allowlist.
+def unbundle_ima_policy(allowlist_bundle, force_verification=False):
+    allowlist_raw = base64.b64decode(allowlist_bundle["ima_policy"])
+
+    # Verify allowlist signatures if set to enforce
+    if force_verification or config.getboolean("cloud_verifier", "require_allow_list_signatures", fallback=False):
```

Comment:
Can you remove `config.getboolean("cloud_verifier", "require_allow_list_signatures", fallback=False)` here and move to the `post()` function where this function is called in the verifier?